### ISSUE TITLE
Run unbuilt version on Windows

### ISSUE THREAD

**Issue by KlaussMC** [eladnava/pixelmate#2]
Hi,

Curious me just discovered your project, and was wondering how I can run this on a windows environment?

I tried starting with `node src/electron.js` from the main folder, but it threw an error saying it couldn't find the module "electron". So (if it's possible) how do I get the program to run under windows (10 x64)?

**Comment by eladnava** [comment]
Hi @KlaussMC,
Nice to connect.

Unfortunately the `adb` binary that comes with Pixelmate is built for OS X (x64) only. Therefore, even if you were to run the app using `electron` on Windows, the `adb` binary would fail to execute.

Is there any reason you prefer using Pixelmate over simply managing your Android storage using the Windows File Explorer?

**Comment by KlaussMC** [comment]
I was originally looking for a graphical software that was ___a)___ open source - hence, customisation, and ___b)___ that was more or less a "simple" project to contribute to, if it was necessary.

Also, it looked like a really cool idea that, with a bit of tinkering, could be brought to life on Windows.

**Comment by KlaussMC** [comment]
I had also just looked over your last comment and realised you mentioned ADB Binary, Could I, by any chance, use the ADB that I have already installed on my computer and made accessible via the command line, in the application without having to ship a binary file?

**Comment by eladnava** [comment]
Absolutely, to run this project on Windows, try this:

1) `git clone https://github.com/eladnava/pixelmate.git`
2) `cd pixelmate`
3) `npm install`
4) `npm install -g electron`
5) Replace `adb` binary in `bin/` with a Windows `adb` binary (from your Android SDK's `platform-tools/` folder)
6) Run `npm run server` in one terminal
7) Run `npm run app` in another terminal and cross your fingers

**Comment by eladnava** [comment]
If you get it to work you can then commit the binary into the repo and add logic in `pathToAdb()` to check the `platform.os` and point to the relevant binary:
https://github.com/eladnava/pixelmate/blob/06b2594dbd7629510a4434b16c58e30bc8a4b116/src/util/adb.js#L201-L203

**Comment by KlaussMC** [comment]
That I will do, thanks so much for your feedback, I hope we can make this a success!

**Comment by KlaussMC** [comment]
Hi, so I did some translation work (darwin -> windows), but I noticed that node calls windows x64 and x86 by the same name, so each distro is named `win32`. I also noticed that you tend to use `import` statements, rather than the usual `require`, Why is that? Also, while i was at school, I couldn't install electron due to proxy issues, so maybe ship the program with electron. You Might also like to do a full test before it's pulled to make sure I haven't broken anything...

**Comment by eladnava** [comment]
1) You can safely attach just an `x64` binary as it's okay not to support the latter for now.
2) `import` is the ES6 way to require modules:
http://exploringjs.com/es6/ch_modules.html#sec_overview-modules
3) The actual `electron` binary is needed just for people who want to build Pixelmate, and contains thousands of files from the Electron framework which would make this repo massive. It therefore does not make sense at this time to commit `electron` sources.

**Comment by KlaussMC** [comment]
Yeah, I might require a bit more research on the import statement, I have never seemed to be able to get it to work. but thanks for your answer.

**Comment by eladnava** [comment]
No problem. If you have the latest [Node.js LTS version](https://nodejs.org/en/download/) it definitely supports `import`. It is pretty much interchangeable with `require`.

**Comment by KlaussMC** [comment]
yeah, I do know a little about them, I've just never had much luck using them sucessfully.

**Comment by KlaussMC** [comment]
I've got the program files for the ADB on windows, should I provide a path directly to the `adb.exe` file or just the folder containing the `adb.exe`? or should I actually ship a `.bin` file? I'm a little stuck but making slow progress.

**Comment by KlaussMC** [comment]
also, when starting the program's server, I get the following error: `TypeError: fs.existsSync is not a function` at `(anonymous function)
C:/Users/************/*******/pixelmate/node_modules/electron/index.js:6`

**Comment by eladnava** [comment]
1) You should place `adb.exe` in the project's `bin/` folder alongside the existing `adb` binary.
2) I have absolutely no idea why you are getting this error. Are you running `npm run server`?

**Comment by KlaussMC** [comment]
Yes, I am running the server, it opens Chrome automatically, loads the page, and displays this error
![screenshot 3](https://user-images.githubusercontent.com/29561775/43834201-a5b651b8-9b50-11e8-9609-d2cf1b9939e7.png)

**Comment by eladnava** [comment]
You're not supposed to open the server URL in Chrome, instead, when you run `npm run app` it will load the URL using Electron and it will then have access to the `fs.exsitsSync` function.

`URL=http://localhost:3000` is the URL in which the `npm run server` command will expose a HTTP server for `npm run app` to connect to.

**Comment by KlaussMC** [comment]
Okay, I believe I understand why that is, however, when I do run `npm run server`, Chrome opens automatically, and it autofocuses too. When I run `npm run app`, I get `'URL' is not recognized as an internal or external command, operable program or batch file.`

**Comment by KlaussMC** [comment]
is there a way to get electron to read a file that contains the string "localhost:3000" for example?

**Comment by KlaussMC** [comment]
![i got somewhere](https://user-images.githubusercontent.com/29561775/43997873-84c2a29e-9e2b-11e8-9dfd-6d2251754cc2.png)

**Comment by eladnava** [comment]
What command are you using to run Electron?

**Comment by KlaussMC** [comment]
`electron . URL=http://localhost:3000`

**Comment by KlaussMC** [comment]
okay, I will try that, I did notice that when I print out `process.env` there is no entry for `URL` even when I start electron with the variable.

**Comment by eladnava** [comment]
Try this:

```
export URL=http://localhost:3000
electron .
```

**Comment by KlaussMC** [comment]
`'export' is not recognized as an internal or external command,
operable program or batch file`

**Comment by eladnava** [comment]
Try this:
```
set URL=http://localhost:3000
electron .
```

**Comment by KlaussMC** [comment]
hol' up, trying git bash...

**Comment by KlaussMC** [comment]
git bash worked, but still no difference

**Comment by eladnava** [comment]
https://github.com/eladnava/pixelmate/issues/2#issuecomment-412316204

**Comment by KlaussMC** [comment]
that works in CMD export in git bash. I would never have thought of that, : /

**Comment by KlaussMC** [comment]
So I got it working right, but I still get that annoying `fs.existsSync` error message, in the app.

**Comment by eladnava** [comment]
Really don't know what would be causing this error. Out of ideas. 😢

**Comment by KlaussMC** [comment]
I tried editing index.js in all my electron installations, that seemed to throw a whole set of other errors, so I doubt it's that, and googling it didn't give any answers, I might have to hardcode some values.

**Comment by KlaussMC** [comment]
in node_modules/electron/index.js, there's a line that looks like this: `if (fs.existsSync(pathFile)) {`, it says this is the line causing the error, so I aliased it, created a variable named `exists` and set it to `fs.existsSync` then rewrote the if block `if (exists(pathFile)) {` and now I get a new error: `TypeError: exists is not a function`. WTH JavaScript!

**Comment by KlaussMC** [comment]
I did a bit of research, and discovered that the issue could potentially be conflict with React.js, i was wondering if you actually get your version to work on Mac, or whether updating some packages spoils it?

**Comment by KlaussMC** [comment]
I worked out what the problem was, I did a check whether i'm on darwin or win32, but because it's running through electron / react, it thinks that the platform is browser... it's a start...

**Comment by KlaussMC** [comment]
![screenshot 8](https://user-images.githubusercontent.com/29561775/44571431-202c4180-a7c4-11e8-9538-d46b2e28be08.png)

I am so excited to tell you that I managed to get the program running on windows, I have done a test to see whether I can navigate/browse different directories, it works, I now need you to confirm that I haven't broken anything for the Mac side.

thanks. more upgrades to come.

**Comment by KlaussMC** [comment]
Excellent, again, I hope it works. and I also realise now that I may have left the console open and a few console.log statements here and there
